src/web/extractors.rs

```diff
@@ -1269,6 +1291,43 @@ impl FromRequest for BsoQueryParams {
                 Some(tags.clone()),
             )
         })?;
+        if params.sort != Sorting::Index {
+            let bound = match params
+                .offset
+                .clone()
+                .unwrap_or(Offset {
+                    timestamp: None,
+                    offset: 0,
+                })
+                .timestamp
+            {
+                None => 0,
+                Some(timestamp) => timestamp.as_i64(),
+            };
+            if bound != 0 {
```

Comment:
Seems like we only need these checks if there's a timestamp, so the above could simplify down to
```suggestion
            if let Some(timestamp) = params.offset.as_ref().and_then(|offset| offset.timestamp) {
                let bound = timestamp.as_i64();
```